git clone https://github.com/pjueon/pykis.git   
cd pykis   
pip3 install -r requirements.txt   
pip3 install pykis

### API 객체 생성

In [36]:
import pykis

In [37]:
# API 사용을 위한 API key정보 및 계좌 정보를 설정합니다. 
# 별도의 파일(json, yaml, xml, etc) 등에 내용을 저장하여 불러오는 것을 추천합니다. 
# key 정보는 절대 외부로 유출되지 않도록 주의하시고, 유출시 즉시 재발급 하시기 바랍니다.  

key_info = {		# KIS Developers 서비스 신청을 통해 발급받은 API key 정보
	"appkey": "PS2Yo0OYRrr7dqtBThWdmiqMj41S9nKefhNP",                  
	"appsecret": "BSk3IctJZpdvxmFI1YWlFuobhJ/82D/W47iyAMHkr7H1Q+ZNk4mdperWmlEw1tG7Ou40aTU8wm/Z4AOsAVCkiqLMKJt1lhEPyn3BmSH/VQUlO5EEuCffJ/ORWyDrnoNYI1kisNOgPkfkZH3wgMBpulpTwpr7BPGPMvwTZDal1GBIgAFQqj4=" 
}

account_info = {	# 사용할 계좌 정보
	"account_code": "50111264",   
	"product_code": "01"             # ex> "01", "22", etc
}

# API 객체 생성 
api = pykis.Api(key_info=key_info, account_info=account_info)

모의 계좌 사용하는 경우

In [38]:
domain = pykis.DomainInfo(kind="virtual")


# API 객체 생성 
api = pykis.Api(key_info=key_info, domain_info=domain, account_info=account_info)

### 사용 계좌 변경

In [39]:
account_info = {    # 사용할 계좌 정보
	"account_code": "50111264",   
	"product_code": "01"             # ex> "01", "22", etc
}

api.set_account(account_info)

## 국내 주식 관련

### 거래 가능 현금 조회

In [41]:
cash = api.get_kr_buyable_cash()
cash

497500000

### 국내 주식 현재가 조회

In [42]:
ticker = "005930"   # 삼성전자 종목코드
price = api.get_kr_current_price(ticker)
price

78100

### 국내 주식 최근 가격 조회 (일/주/월 OHLCV)

In [43]:
# 최근 30 일/주/월 OHLCV 데이터를 DataFrame으로 반환
ticker = "005930"   # 삼성전자 종목코드
time_unit = "D"     # 기간 분류 코드 (D/day-일, W/week-주, M/month-월), 기본값 "D"
ohlcv = api.get_kr_ohlcv(ticker, time_unit)
ohlcv

Open   High    Low  Close    Volume
Date                                            
2024-05-22  78100  78200  77300  78000   5865386
2024-05-21  78500  79000  78200  78400  13690583
2024-05-20  78100  79100  77900  78900  19456783
2024-05-17  78600  78800  77200  77400  15918201
2024-05-16  80200  80300  78100  78200  20989778
2024-05-14  78600  78800  77900  78300  11763992
2024-05-13  79400  79900  77600  78400  18652344
2024-05-10  80400  81100  78900  79200  16976124
2024-05-09  81100  81500  79700  79700  18759935
2024-05-08  80800  81400  80500  81300  13089576
2024-05-07  79600  81300  79400  81300  26238868
2024-05-03  79000  79000  77500  77600  13151889
2024-05-02  77600  78600  77300  78000  18900640
2024-04-30  77000  78500  76600  77500  19007007
2024-04-29  77400  77600  76200  76700  14664474
2024-04-26  77800  77900  76500  76700  12755629
2024-04-25  77300  77500  76300  76300  15549134
2024-04-24  77500  78800  77200  78600  22166150
2024-04-23  76400  76800  75500  75500  18717699
2024-04-22  77400  77500  75100  76100  30469477
2024-04-19  78300  78700  76300  77600  31317563
2024-04-18  78800  80100  78300  79600  21370190
2024-04-17  80700  80800  78900  78900  22611631
2024-04-16  81200  81300  79400  80000  31949845
2024-04-15  82900  83200  81200  82200  26663772
2024-04-12  84700  84900  83200  83700  17061770
2024-04-11  83200  84700  82500  84100  25538009
2024-04-09  84500  84900  83100  83600  23725956
2024-04-08  85200  86000  84500  84500  18953232
2024-04-05  84500  85000  83800  84500  18883752

### 국내 주식 하한가 조회

In [44]:
ticker = "005930"   # 삼성전자 종목코드
price = api.get_kr_min_price(ticker)
price

54900

### 국내 주식 상한가 조회

In [45]:
ticker = "005930"   # 삼성전자 종목코드
price = api.get_kr_max_price(ticker)
price

101900

### 국내 주식 잔고 조회

In [46]:
# DataFrame 형태로 국내 주식 잔고 반환 
stocks_kr = api.get_kr_stock_balance()
stocks_kr

Empty DataFrame
Columns: []
Index: []

### 국내 주식 총 예수금 조회

In [47]:
deposit = api.get_kr_deposit()
deposit

500000000

### 국내 주식 매수 주문(장 열린 시간에 가능)

In [54]:
ticker = "005930"   # 삼성전자 종목코드
price = 100000      # 매수 가격 예시. 가격이 0 이하인 경우 시장가로 매수
amount = 1          # 주문 수량

# 삼성전자 1주를 지정가로 매수 주문 
api.buy_kr_stock(ticker, amount, price=price)

{'KRX_FWDG_ORD_ORGNO': '00950', 'ODNO': '4302', 'ORD_TMD': '101750'}

여기서 'KRX_FWDG_ORD_ORGNO' 은 한국거래소에 전달된 주문 기관 번호를 의미함. 00950은 주문을 처리한 기관의 식별 코드.    
'ODNO'는 주문 번호로, 개별 주문을 식별하는 고유번호   
'ORD_TMD'는 주문 시간을 나타냄(100740은 오전 10시 07분 40초에 주문이 처리되었음을 의미)

### 국내 주식 매도 주문

In [49]:
ticker = "005930"   # 삼성전자 종목코드
price = 100000      # 매도 가격 예시. 가격이 0 이하인 경우 시장가로 매도
amount = 1          # 주문 수량

# 삼성전자 1주를 지정가로 매도 주문 
api.sell_kr_stock(ticker, amount, price=price)

{'KRX_FWDG_ORD_ORGNO': '00950', 'ODNO': '3865', 'ORD_TMD': '100756'}

### 정정/취소 가능한 국내 주식 주문 조회 
- 모의투자에서는 불가한 기능

In [50]:
# 정정/취소 가능한 국내 주식 주문을 DataFrame으로 반환
orders = api.get_kr_orders()

RuntimeError: http response: 200, return code: 1. msg: 모의투자에서는 해당업무가 제공되지 않습니다.                                    

### 미체결 국내 주식 주문 취소 
- api.get_kr_orders 이게 모의투자에선 불가한 기능이라 order_number 확인 불가하여 미체결 주문 취소도 불가능함

In [51]:
# order_number: 주문 번호. api.get_kr_orders 통해 확인 가능.
# amount: 취소할 주문 수량. 지정하지 않은 경우 잔량 전부 취소.
api.cancel_kr_order(order_number, amount)

NameError: name 'order_number' is not defined

### 모든 미체결 국내 주식 주문 취소 
- 모의투자에서는 불가

In [52]:
api.cancel_all_kr_orders()

RuntimeError: http response: 200, return code: 1. msg: 모의투자에서는 해당업무가 제공되지 않습니다.                                    

### 주식 주문 정정 
- 모의투자에서는 불가

In [53]:
# order_number: 주문 번호. api.get_kr_orders 통해 확인 가능.
# price: 정정할 1주당 가격.
# amount: 정정할 주문 수량. 지정하지 않은 경우 잔량 전부 정정.
api.revise_kr_order(order_number, price, amount)

NameError: name 'order_number' is not defined